apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.js

```diff
@@ -17,136 +15,125 @@ import {
 } from '../../lib/editors';
 import { encodePath } from '../../lib/uri-util';
 import { getCanonicalHost } from '../../lib/url';
+import { IdSchema } from '../../lib/db-types';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-router.get('/', function (req, res, next) {
-  debug('GET /');
-  async.series(
-    [
-      (callback) => {
-        debug('query short_names');
-        sqldb.queryOneRow(sql.short_names, { course_id: res.locals.course.id }, (err, result) => {
-          if (ERR(err, callback)) return;
-          res.locals.short_names = result.rows[0].short_names;
-          callback(null);
-        });
-      },
-    ],
-    function (err) {
-      if (ERR(err, next)) return;
-      const host = getCanonicalHost(req);
-      res.locals.studentLink = new URL(
-        res.locals.plainUrlPrefix + '/course_instance/' + res.locals.course_instance.id,
-        host,
-      ).href;
-      res.locals.studentLinkQRCode = new QR({
-        content: res.locals.studentLink,
-        width: 512,
-        height: 512,
-      }).svg();
-      res.locals.infoCourseInstancePath = encodePath(
-        path.join(
-          'courseInstances',
-          res.locals.course_instance.short_name,
-          'infoCourseInstance.json',
-        ),
-      );
-      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-    },
-  );
-});
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    res.locals.short_names = await sqldb.queryRow(
+      sql.short_names,
+      { course_id: res.locals.course.id },
+      z.array(z.string()),
```

Comment:
This is using an `array_agg` behind the scenes. Rather than do that, how about we just return a standard one-row-per-name and change the type here to be `z.string()`?

Reply:
That makes sense. Updated to this in [6a9dc74](https://github.com/PrairieLearn/PrairieLearn/pull/9591/commits/6a9dc74dfafc5b4afe9791e93e16b59f1471c3dc).